src/topology/algebra/order/lower_topology.lean

```diff
@@ -0,0 +1,144 @@
+/-
+Copyright (c) 2022 Christopher Hoskin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Christopher Hoskin
+-/
+import order.upper_lower
+import topology.separation
+
+/-!
+# Lower topology
+
+This file introduces the lower topology on a preorder. It is shown that the lower topology on a
+partial order is T₀ and the complements of the upper closures of finite subsets form a basis.
+
+## Implementation notes
+
+Approach inspired by `order_topology`.
+
+## References
+
+* [Gierz et al, A Compendium of Continuous Lattices][GierzEtAl1980]
+
+## Tags
+
+lower topology, preorder
+-/
+
+universe u
+
+variable (α : Type u)
+
+open set topological_space
+
+section preorder
+
+variable [preorder α]
+
+/--
+Type synonym for a preorder equipped with the lower topology
+-/
+@[derive preorder, nolint unused_arguments]
```

Comment:
I would suggest not having `[preorder α]` be an argument to `with_lower_topology` for performance reasons
```suggestion
```